### DISCUSSION TITLE
Long term caching (> 31 days) for responses from serverless functions

### DISCUSSION THREAD

**Discussion by danmichaelo** [vercel/community#798] [Help] [answered]
Hi! We have a serverless API endpoint that generates content that takes a while to generate (5-10 seconds), but can basically be cached indefinitely since there will always be the same output for the same query string parameters. I have two questions:

* If I set `Cache-Control` to something longer than 31 days, is it correct that Vercel will still only cache it for (up to) 31 days?
* Are all caches flushed when the app is re-deployed?

For this scenario, would I be better off using a more long-term cache (say an S3 bucket) in addition to Vercel's cache?

**Comment by amyegan** [comment, marked as answer]
Hi @danmichaelo! The [cache limit ](https://vercel.com/docs/concepts/edge-network/caching#limits) is 31 days so setting it to something longer won't keep it around indefinitely. And new deployments do [invalidate the cache](https://vercel.com/docs/concepts/edge-network/frequently-asked-questions#how-do-i-purge-the-cdn-cache).

Using [`stale-while-revalidate`](https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching#stale-while-revalidate) looks like a good solution for this scenario because it lets you serve from cache while simultaneously updating the cache in the background with the response from your Serverless Function.
